cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/TimeRangeConstraint.java

```diff
@@ -34,6 +34,7 @@
   private transient int endHour;
   private transient int endMinute;
 
+  private transient boolean isStartTimeSmaller;
```

Comment:
this is not modified only in the constructor.
In the case of gson deserialization, this is computed and updated after object creation.